### ISSUE TITLE
Disable CRIUSecProvider when specific option is set during checkpoint

### ISSUE THREAD

**Issue by jasonkatonica** [eclipse-openj9/openj9#18422]
Whenever performing a checkpoint using openj9 we are typically able to do something like the following:

```
java -XX:+EnableCRIUSupport  QuickTest
```

When called by default all security providers are removed from the runtime environment and the `CRIUSecProvider` is inserted into the runtime. This approach provides a safer approach to locking down what cryptography is allowed during the checkpoint phase and additionally forces MessageDigests supported to be reset and SHA1PRNG instances to reset their seeds that are in use.

We would like to introduce a new option named `-XX:+EnableFullSecurityEnv`. Using this option all checkpoint related security considerations shift from the responsibility of a Java runtime to the application owner and deployment model in use:

```
java -XX:+EnableCRIUSupport -XX:+EnableFullSecurityEnv QuickTest
```

With this option we no longer make use of the `CRIUSecProvider` in any fashion. In this case we would use the default set of providers and algorithms to enable full functionality prior to checkpoint.

There have been a few use cases identified so far for this option. Both of these cases assume that the application designers are able to correctly handle some CRIU specific concerns such as forcing the use of entropy sources correctly for random numbers in use, zeroing memory, de-referencing sensitive objects, protected CRIU files at an OS level using correct permissions, and other security related tasks assigned with general protection of a file at rest. The main uses cases are as follows:
- Users are not making use of "real" sensitive fields prior to checkpoint. In this case they no longer need any additional processing in the CRIU provider, nor restrictions on algorithms, since they are not making use of real sensitive information prior to checkpoint. This use case is a real world use case in place for spring boot applications running within liberty. 
- Deployments may choose to make use of disk encryption or any other encryption technologies to secure checkpoint files at rest. In this case all values are encrypted and cannot be exposed by design to others.

Both of these use cases can also be combined.

This option also provides the following additional advantages over the use of the CRIUSecProvider strategy outlined above:

- Resetting the state of algorithms, setting seeds of random generators, and clearing sensitive values may or may not be a desired behavior for the users applications.
- The CRIUSecProvider has very limited functionality that is currently supported. Over time this will require more and more additional algorithms to be developed and deployed on a case by case basis for a given CRIU use case.
- Other solutions that are are available today from other runtimes support this type of hands off approach.

We propose to implement this option as a non default option due to the risks of using such an approach.

**Comment by keithc-ca** [comment]
I'm still not a fan of the option name, `-XX:+EnableFullSecurityEnv`; reading this made me wonder whether `-XX:-CRIUSecProvider` or `-XX:+DisableCRIUSecProvider` would be acceptable. Both include "CRIU" making it clear that it relates to CRIU checkpoints and both make it clear that the CRIU security provider is not to be used. Thoughts?

**Comment by tajila** [comment]
My vote is for `-XX:-CRIUSecProvider`. I like that the `-XX:-` indicates the removal of a feature so its clear that users are disabling something.

**Comment by jasonkatonica** [comment]
Yes agreed that the `-XX:+EnableFullSecurityEnv` name is not the greatest. I would be fine using the option name `-XX:-CRIUSecProvider` as this flag will specifically mean to either use this provider or not during CRIU processing.

**Comment by WilburZjh** [comment]
Updating the code accordingly, will push soon

**Comment by keithc-ca** [comment]
While reviewing https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/689 I wondered whether the new option should also allow acceleration of the extra algorithms via OpenSSL. Opinions?
